src/OmniSharp.Roslyn.CSharp/Services/Decompilation/AssemblyResolver.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.PortableExecutable;
+using ICSharpCode.Decompiler.Metadata;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+
+namespace OmniSharp.Roslyn.CSharp.Services.Decompilation
+{
+    internal class AssemblyResolver : IAssemblyResolver
+    {
+        private readonly Compilation _parentCompilation;
+        private readonly Dictionary<string, List<IAssemblySymbol>> _cache = new Dictionary<string, List<IAssemblySymbol>>();
+        private readonly ILogger _logger;
+
+        public AssemblyResolver(Compilation parentCompilation, ILoggerFactory loggerFactory)
+        {
+            _parentCompilation = parentCompilation;
+            _logger = loggerFactory.CreateLogger<AssemblyResolver>();
+            BuildReferenceCache();
+
+            void BuildReferenceCache()
```

Comment:
nit, local function doesn't bring any value here.

Reply:
this is internalized from here http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.EditorFeatures/DecompiledSource/AssemblyResolver.cs,29

I would like to keep all the code intact  (except the one place we needed a change)